NCKU-SE-DP/pricetracker-percykugood
# Fix/integration test

## Purpose
The purpose of this pull request is to resolve CI testing issues and ensuring stable and reliable integration testing across the project.

## Changes
1. Test Setup Improvements :
   - Delete `test.db` file before each test run to ensure a clean state.
   - Updated `setup_and_teardown` fixture to handle database file deletion and recreation.

2. Wrap OpenAI API Request :
   - Wrapped the OpenAI API request in a separate function and mocked it in tests to avoid using actual API keys.

3. Set PowerShell Execution Policy :
   - Added a step in the GitHub Actions workflow to set the PowerShell execution policy to `Unrestricte
